### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by AlexNDRmac**
Hi @NoiseByNorthwest. This is first part of #180

This is base workflow which includes cross-php version builds for mac/linux OS.

## Features

- Now each build will store build artifact to workflow summary. It's very useful, users don't need to compile extension - just download compiled and use!
- If smth goes wrong - CI workflow have an action step to store logs for the failed step which helps to sort out with possible mistakes.
- CI have step to check the release notes parser - this is useful to check if everything is OK with release notes before creating new Release.
- Created CHANGELOG with all the history of changes.
- Created release notes parser from changelog to simplify Release notes creation. Now, all you need - just push the `tag` with for new version or simply create new release with empty release notes body - Release action will do everything else. Release action will run CI to ensure that everything is OK, then download all the build artifacts, parse release notes for the latest TAG in CHANGELOG.md and creates Release with this content and all the build artifacts (extensions for all PHP version and all supported OS)

## What i missed

- I'm not removed Travis workflow file - please remove it yourself after github actions successful merge
- I'm do not add new version to CHANGELOG, only add changes to **Unreleased** section. You can easily create new version yourself when you decide - just move everything from **Unreleased** to new version section and change link to compare changes in H2 markdown tag

Thanks

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      - name: Create ZIP archive with build artifact
+        run: |
+          zip -rvj ${{ steps.setup-artifact.outputs.spx_file_name }}.zip \
+            ./modules/spx.so LICENSE $ZIP_EXCLUDE
+          zip -rv ${{ steps.setup-artifact.outputs.spx_file_name }}.zip \
+            ./assets $ZIP_EXCLUDE
+        env:
+          ZIP_EXCLUDE: -x ".*" -x "__MACOSX" -x "*.DS_Store"
+
+      - name: Check Release notes
```
Comment (NoiseByNorthwest) [inline .github/workflows/main.yml:97]: What is the point of this step in this context ?

Comment (NoiseByNorthwest) [reply .github/workflows/main.yml:97]: OK, to check that the script do not fail before entering the release process.

**Comment by NoiseByNorthwest** [comment]
@AlexNDRmac Thanks for your work, LGTM.

Do I have to merge first in order to enable github actions ?

**Review by NoiseByNorthwest — APPROVED** [review]

**Comment by AlexNDRmac** [comment]
@NoiseByNorthwest - yes, to enable actions, you should merge first. After merge, actions will start and status badge also should retrieve status automatically.

**Pull request merged**